x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -164,41 +170,25 @@ public Realm realm(String name) {
-        List<String> kerberosRealmNames = new ArrayList<>();
         Map<String, Set<String>> nameToRealmIdentifier = new HashMap<>();
         Map<Integer, Set<String>> orderToRealmName = new HashMap<>();
-        for (RealmConfig.RealmIdentifier identifier: realmsSettings.keySet()) {
-            Realm.Factory factory = factories.get(identifier.getType());
-            if (factory == null) {
-                throw new IllegalArgumentException("unknown realm type [" + identifier.getType() + "] for realm [" + identifier + "]");
-            }
-            RealmConfig config = new RealmConfig(identifier, settings, env, threadContext);
+        for (RealmConfig config: realmConfigs) {
+            Realm.Factory factory = factories.get(config.identifier().getType());
+            assert factory != null : "unknown realm type [" + config.identifier().getType() + "]";
             if (config.enabled() == false) {
                 if (logger.isDebugEnabled()) {
-                    logger.debug("realm [{}] is disabled", identifier);
+                    logger.debug("realm [{}] is disabled", config.identifier());
                 }
                 continue;
             }
-            if (FileRealmSettings.TYPE.equals(identifier.getType()) || NativeRealmSettings.TYPE.equals(identifier.getType())) {
-                // this is an internal realm factory, let's make sure we didn't already registered one
-                // (there can only be one instance of an internal realm)
-                if (internalTypes.contains(identifier.getType())) {
-                    throw new IllegalArgumentException("multiple [" + identifier.getType() + "] realms are configured. ["
-                            + identifier.getType() + "] is an internal realm and therefore there can only be one such realm configured");
-                }
-                internalTypes.add(identifier.getType());
-            }
-            if (KerberosRealmSettings.TYPE.equals(identifier.getType())) {
-                kerberosRealmNames.add(identifier.getName());
-                if (kerberosRealmNames.size() > 1) {
-                    throw new IllegalArgumentException("multiple realms " + kerberosRealmNames.toString() + " configured of type ["
-                        + identifier.getType() + "], [" + identifier.getType() + "] can only have one such realm " +
-                        "configured");
-                }
+            // Ensure we always have some values reserved at each end of integer's range
+            // so they can be used for adding file and native realms
+            if (config.order() <= REALM_ORDER_LOWER_BOUND || config.order() >= REALM_ORDER_UPPER_BOUND) {
+                throw new IllegalArgumentException(String.format(Locale.ROOT,
+                    "realm order value must be greater than [%s] and smaller than [%s], but got [%s] for [%s]",
+                    REALM_ORDER_LOWER_BOUND, REALM_ORDER_UPPER_BOUND, config.order(), config.identifier()));
```

Comment:
**Nit**: It can be done without imposing realm order bounds, similarly to the reserved realm case. If we're introducing realm bounds I would prefer something like 0-1000. Using large values like these is my least preferred option, because they seem arbitrary and scary in the errors.

Reply:
This is not a nit, but instead a good advice. Thanks! I removed the bounds and took the similar approach as how reserved realm is handled.